sycl/source/detail/pi_opencl.cpp

```diff
@@ -78,6 +78,43 @@ pi_result OCL(piextDeviceSelectBinary)(
   return PI_SUCCESS;
 }
 
+pi_result OCL(piQueueCreate)(pi_context context, pi_device device,
+                             pi_queue_properties properties, pi_queue *queue) {
+  size_t devVerSize;
+  cl_int ret_err = clGetDeviceInfo(pi_cast<cl_device_id>(device),
+                                   CL_DEVICE_VERSION, 0, NULL, &devVerSize);
+
+  if (ret_err != CL_SUCCESS)
+    return pi_cast<pi_result>(ret_err);
+
+  std::string devVer(devVerSize, '\0');
+  clGetDeviceInfo(pi_cast<cl_device_id>(device), CL_DEVICE_VERSION, devVerSize,
+                  &devVer.front(), NULL);
```

Comment:
why is this per-device and not per-platform? clGetPlatformInfo has a CL_PLATFORM_VERSION which should report the OpenCL version supported for the OpenCL runtime in the host, so checking it per platform should be enough...

Reply:
You are correct, thank you for the catch. Will try to fix this in the next commit! Sorry for the trouble.